### DISCUSSION TITLE
Question on Tutorials - https://supabase.io/docs/guides/with-flutter

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#2785] [Questions] [answered]
Could be a newbie question, I am just wondering where is the recoverSupabaseSession() method defined in this tutorial. I see the call made inside of the splash_page.dart file, but do not see the definition anywhere.
![image](https://user-images.githubusercontent.com/33538243/128761876-c773e75c-3059-49b7-9e51-3d98457352af.png)

**Comment by dshukertjr** [comment, marked as answer]
Hi @sumchans!

`recoverSupabaseSession()` is coming from `AuhtState` class, which inherits `SupabaseAuthState` class defined [here](https://github.com/supabase/supabase-flutter/blob/main/lib/src/supabase_auth_state.dart). Notice that `_SplashPageState` does not inherit `State` class like usual stateful widget definition, but is inheriting `AuthState` class instead. This `AuthState` class listens to app's lifecycle events and takes appropriate auth actions like signing users in from third party OAuth or Magic link sign in.

**Reply by sumchans** [reply]
Oh okay, Where do I go to see all the methods and properties exposed by the AuthState Class. Is there a documentation anywhere?

**Reply by dshukertjr** [reply]
@sumchans 
I actually updated my answer a bit. `recoverSupabaseSession ()` comes from `SupabaseAuthState`, which `AuthState` inherits from. 

Unfortunately, the only way to view all the methods that components from `supabase_flutter` exposes is by directly viewing the code or by viewing the [example apps](https://github.com/phamhieu/supabase-flutter-demo). `supabase_flutter` is still at its early stage, and it is maintained by the community, so we are working hard to bringing better developer experience! Hopefully, we will have a nice set of docs in the near future.

**Reply by sumchans** [reply]
Not a problem. Thanks Tyler.

**Reply by sumchans** [reply]
Does the latest supabase_flutter package support phone authentication? I am trying this code, it doesn't seem to work. 
![image](https://user-images.githubusercontent.com/33538243/128794450-f5bca1e1-fa8f-41d1-a36f-5dff7afbf1f7.png)

Would you be able to include some snippets on how to make it work with Flutter?

**Reply by dshukertjr** [reply]
@sumchans 
Sorry for the confusion and lack of documentation and thank you so much for the feedback. We will include a sample code snippet in the readme for other developers to see as well. 

Because `signUp` uses positional arguments, we didn't include `phone` signup in the `signUp()` method, but created a separate `signUpWithPhone ()` method for that. We would like to merge these two methods in the future just like how the JavaScript SDK does it. 

`signIn` on the other hand works the same way as Javascript client, and you can just pass a phone number as a parameter.

**Reply by sumchans** [reply]
So is the signUpWithPhone available to use, I tried that as well - this is the message I see on my IDE - **The method 'signUpWithPhone' isn't defined for the type 'GoTrueClient'.
Try correcting the name to the name of an existing method, or defining a method named 'signUpWithPhone'.**. Yeah a proper documentation atleast on pub.dev where the package is listed would be great.

Thanks Tyler

**Reply by sumchans** [reply]
Sorry my mistake, I see it now. My supabase package was not upgraded to the latest version.

**Reply by sumchans** [reply]
Following the above supabase flutter tutorial, I am getting an Unhandled exception -  Hive error when running it on the iOS simulator. It happens on the splashPage class. If I comment out recoverSupabaseSession() method, I don't get any error. Here is the snip of the error -
![Screen Shot 2021-08-09 at 8 12 10 PM](https://user-images.githubusercontent.com/33538243/128802689-63113515-2d21-447d-b72f-8673837412d0.png)

**Reply by dshukertjr** [reply]
@sumchans 
Oops, sorry to hear that. Let me look through it and get back to you! It might take several hours for me to take a look and get back to you as I am currently at my other job, but I will get to it ASAP. I am so sorry for a rough experience around phone auth. We will try to do better in the future!

**Reply by sumchans** [reply]
Sure no problem, Thanks Tyler.

**Reply by sumchans** [reply]
Hi Tyler, Sorry if I am bugging you too much, did you have any chance to look into this issue?

**Reply by dshukertjr** [reply]
So sorry for the late action, had a busy day yesterday. I will start looking into it right now. Thanks for baring with me!

**Reply by sumchans** [reply]
No problems, thanks again.

**Reply by dshukertjr** [reply]
@sumchans 
I tested it out and was able to sign in user without facing the error you are experiencing. Do you have your source code uploaded to a public repo? Maybe I could take a look at that and try it out on my end to see if it works.

**Reply by sumchans** [reply]
@dshukertjr, I just copied the main files from my project to this location https://github.com/sumchans/Bol. It s almost like how you have setup in the tutorial, just made a few changes here and there. Let me know if you need anything more. 

Also as I mentioned if I comment out the recoverSupabaseSession() method call the code runs.

Thank you for looking into this.

**Reply by sumchans** [reply]
@dshukertjr Ignore the above. Previously I was modifying an existing project. Since you mentioned that you didn't see the same error, I thought of testing it out on a brand new project. It runs without issues.

**Reply by dshukertjr** [reply]
@sumchans 
Glad everything worked out! If you ever have any question, you can always open another discussion question and someone should be able to answer it shortly!

**Reply by sumchans** [reply]
Sure will do. Thanks Tyler.

**Reply by joewinke** [reply]
hey @dshukertjr is there any way to try the example apps without installing them? I am new and want to learn and get started but I thought being able to use the different ones first and see the one I enjoy the most would be the best to learn with. The installations look non-trivial. Thanks!

**Reply by dshukertjr** [reply]
Hi @joewinke!

The example app supports Flutter web, which does not require installing any apps on your phone, but would that work?

**Reply by joewinke** [reply]
@dshukertjr what do you mean? Is there a link to where I can load the different examples w/o installing? I would prefer to do it in desktop

**Reply by dshukertjr** [reply]
@joewinke 

[Here](https://github.com/supasquad/supabase-flutter-quickstart) is the repo of the example app in the quick starter guide. You can clone this to your laptop, and run it on Flutter web by running the following command:

```bash
flutter run -d web-server --web-hostname localhost --web-port 3000
```

This will open up the example app on your browser, and does not involve you installing anything on your phone. Is this what you mean by "not wanting to installing anything"? If not, could you clarify what exactly you mean?
